packages/e2e-tests/specs/experiments/navigation-editor.test.js

```diff
@@ -679,4 +680,143 @@ describe.skip( 'Navigation editor', () => {
 			expect( lastItemAttributes.isTopLevelLink ).toBeTruthy();
 		} );
 	} );
+	describe( 'Delete menu button', () => {
+		afterEach( async () => {
+			await setBrowserViewport( 'large' );
+		} );
+		it.each( [ 'large', 'small' ] )(
+			`should retain menu when confirmation is canceled and the viewport is %s`,
+			async ( viewport ) => {
+				const menuName = 'Menu delete test';
+				await createMenu( { name: menuName }, menuItemsFixture );
+				await visitNavigationEditor();
+				await setBrowserViewport( viewport );
+				// Wait for the header to show the menu name.
+				await page.waitForXPath(
+					`//h2[contains(text(), "${ menuName }")]`
+				);
+
+				if ( viewport === 'small' ) {
+					const openSettingsSidebar = await page.waitForXPath(
+						'//button[@aria-label="Settings"][@aria-expanded="false"]'
+					);
+					await openSettingsSidebar.click();
+				}
+
+				const deleteMenuButton = await page.waitForXPath(
+					'//button[text()="Delete menu"]'
+				);
+				await deleteMenuButton.click();
+
+				const cancelButton = await page.waitForXPath(
+					'//*[@role="dialog"]//button[text()="Cancel"]'
+				);
+				await cancelButton.click();
+
+				const menuActionsDropdown = await page.waitForXPath(
+					`//*[contains(@class,"edit-navigation-menu-actions")]//h2[text()="${ menuName }"]`
```

Comment:
I was about to suggest that, instead of a classname, we'd look for the dropdown's toggle button instead (a `button` with an `aria-label` of `Switch menu`, [according to the code](https://github.com/WordPress/gutenberg/blob/208e5bc2f3f81fb576db2ac2f093e5e944e87051/packages/edit-navigation/src/components/header/menu-actions.js#L56)), but as I inspected the page locally, I realised the aria-label was not there.

That's because the `label` should be applied as a prop of `DropdownMenu` (i.e `<DropdownMenu label={ __('Switch Menu') } ....`), and not as part of `toggleProps`.

Once this PR is merged, we could open a follow-up PR where we fix this bug and we update the e2e introduced here to use the selector that I suggested above — WDYT ?

Reply:
Oh awesome! I was also looking for a non-class approach to this and didn't see anything I could point to... Thanks for sharing the context on why that was happening!

A followup PR sounds great, I'll put it on my to-do list once this merges!